src/main/scala/mesosphere/marathon/core/health/impl/HealthCheckWorkerActor.scala

```diff
@@ -163,6 +154,32 @@ class HealthCheckWorkerActor(implicit mat: Materializer) extends Actor with Stri
       }
     }
   }
+
+  def singleRequest(httpRequest: HttpRequest)(implicit mat: Materializer): Future[HttpResponse] = {
+    if (httpRequest.uri.scheme.equalsIgnoreCase("https")) {
```

Comment:
Do you think it makes sense to replace this check with an additional argument `isSecure: bool`?
Since we are calling the `singleRequest` method from `http` and `https` we already know in advance if it's HTTPS or not.

Reply:
I don't know, additional argument will create more opportunities to mess something up :)